### ISSUE TITLE
Clarify deployment types in `deploy` command

### ISSUE THREAD

**Issue by spalladino** [OpenZeppelin/openzeppelin-sdk#1506]
Running `oz deploy` on 2.8 prompts for the deployment type, listing regular, upgradeable, or minimal as options:
```
$ npx oz deploy
? Choose the kind of deployment (Use arrow keys)
❯ regular 
  upgradeable 
  minimal 
```

Since we are making `deploy` the default command, we should provide some additional info as to what each option means. Consider adding a one-sentence hint (and potentially a link to the docs) for each option, which is shown when the user has the selection arrow placed on it. 

For example:
```
$ npx oz deploy
? Choose the kind of deployment (Use arrow keys)
❯ regular 
  upgradeable 
  minimal 
A regular deployment will create an instance of your contract on the blockchain and call its constructor.
```
```
$ npx oz deploy
? Choose the kind of deployment (Use arrow keys)
  regular 
❯ upgradeable 
  minimal 
An upgradeable deployment requires that your contracts use initializers, and have slightly higher gas cost, but will allow you to change the code of your contract (while preserving its address, state, and funds) by running oz upgrade, using a proxy behind the scenes.
```
```
$ npx oz deploy
? Choose the kind of deployment (Use arrow keys)
  regular 
  upgradeable 
❯  minimal 
A minimal proxy deployment (EIP1167) is ideal when you need to deploy multiple instances of the same contract, as the first deployment is slightly more expensive, but all subsequent ones are much cheaper.
```

**Comment by spalladino** [comment]
Note that inquirer does not support this feature out of the box. I've [sent a PR](https://github.com/SBoudrias/Inquirer.js/pull/905) to add it, though if it does not get accepted, we could roll out our own prompt kind with support for this.

**Comment by frangio** [comment]
Fixed in #1510
